### PR TITLE
ADOBE_materials_clearcoat_tint

### PR DESCRIPTION
**Pull request by MiiBond**
If there's enough interest in this, it would be nice to make it KHR. I realize that we have a lot of in-progress KHR extensions at the moment though :)

### CODE REVIEW

**Comment by donmccurdy** [comment]
My preference would be to start with an ADOBE_ vendor prefix on this for now, if that's ok. As the in-progress KHR PBR extensions are released, I'm sure we will get feedback on what features people are hoping for next, or how much complexity is too much. 

Is this complete and implemented from your perspective? If so I am happy to merge the PR.

**Comment by MiiBond** [comment]
This is implemented in Adobe Dimension (though not in a release build yet). Yes, we can merge it.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            "description": "The transmittance of the clearcoat layer.",
+            "default": [ 1.0, 1.0, 1.0 ],
+            "minItems": 3,
+            "maxItems": 3,
+            "gltf_detailedDescription": "The colour of light allowed to be transmitted through the clearcoat layer of the material. A value of black means no light passes through. A value of white means all light passes through."
+        },
+        "clearcoatTintTexture": {
+            "allOf": [ { "$ref": "textureInfo.schema.json" } ],
+            "description": "The clearcoat layer tint texture.",
+            "gltf_detailedDescription": "The clearcoat layer tint texture. These values are sampled from the R channel.  The values are linear.  Assume white colour if no texture is supplied."
```
Comment (emackey) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json:24]: Should this texture be sRGB, not linear R?

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### Clearcoat Tint
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+As with the core glTF 2.0 spec, all coloured textures are assumed to be sRGB and all coloured factors are assumed to be linear.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatTintFactor**               | `number[4]`                                                                        | The clearcoat tint's colour factor.         | No, default: `[1.0, 1.0, 1.0, 1.0]`   |
```
Comment (emackey) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:59]: The array length is 3.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            "items": {
+                "type": "number",
+                "minimum": 0.0,
+                "maximum": 1.0
+            },
+            "description": "The transmittance of the clearcoat layer.",
+            "default": [ 1.0, 1.0, 1.0 ],
+            "minItems": 3,
+            "maxItems": 3,
+            "gltf_detailedDescription": "The colour of light allowed to be transmitted through the clearcoat layer of the material. A value of black means no light passes through. A value of white means all light passes through."
```
Comment (emackey) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json:19]: Like other glTF factors, this should be expressed as linear, even though the associated texture is sRGB.

**Comment by emackey** [comment]
If this were to become KHR_, we might consider getting rid of the word "clear" since this is tinted now.  So, KHR_material_coat or similar.  Might also combine with IOR & specular.

**Comment by donmccurdy** [comment]
> ... we might consider getting rid of the word "clear" since this is tinted now. So, KHR_material_coat or similar ...

I personally tend to prefer an explicit relationship to the `_materials_clearcoat` extension, despite that detail. Extending `_materials_clearcoat` with `_materials_clearcoat_*` is more intuitive.

I would note that "tint" often means mixing white with a color. In Blender, for example, sheen tint is a 1-channel value, not a 3-channel value, determining the sheen's color as a mix of baseColor and white. tint=1 would mean the sheen is fully white, tint=0 would mean the sheen takes the base color. Similar with Blender's "Specular Tint". I don't feel so strongly about this as to suggest that Dimension needs to change it's vendor implementation, but FWIW, `_materials_clearcoat_color` might be more correct? On the other hand, [Standard Surface](https://autodesk.github.io/standard-surface/#closures/coating) uses "tint" in the way you're using it here.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json (hunk 1 above)
Comment (MiiBond) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json]: Indeed. Thank you!

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md (hunk 2 above)
Comment (MiiBond) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md]: Indeed. Thank you!

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json (hunk 3 above)
Comment (MiiBond) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json]: Indeed. Thank you!

**Comment by MiiBond** [comment]
Thanks, guys. I addressed the small issues mentioned.
As for "color" VS "tint", I'm okay with either. It would be nice for this to align with a future KHR extension so I'm open to changing it based on what the group thinks. This is already implemented in a prerelease build of Dimension but we won't ship a production build with this in it for a while so we have time to change it (and maybe even transition to a KHR_ prefix :) )

**Comment by MiiBond** [comment]
Hey guys.
Just found this old PR for an ADOBE_ extension that isn't merged. Is there any chance we could get an approval on it and have it merged?

**Review by emackey — APPROVED** [review]
Given that this is a vendor extension and has been in active use for a while, it seems fine to me. Probably there will be a KHR version of this, but that's outside scope for this PR.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+All implementations should use the same calculations for the BRDF inputs. Implementations of the BRDF itself can vary based on device performance and resource constraints. See [Appendix B](/specification/2.0/README.md#appendix-b-brdf-implementation) for more details on the BRDF calculations.
+
+As with the core glTF 2.0 spec, all coloured textures are assumed to be sRGB and all coloured factors are assumed to be linear.
+
+|                                  | Type                                                                            | Description                            | Required             |
+|----------------------------------|---------------------------------------------------------------------------------|----------------------------------------|----------------------|
+|**clearcoatTintFactor**               | `number[3]`                                                                        | The clearcoat tint's colour factor.         | No, default: `[1.0, 1.0, 1.0]`   |
+|**clearcoatTintTexture**              | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo)             | The clearcoat layer's tint texture. | No                   |
```
Comment (lexaknyazev) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:60]: The link to the base spec should be updated.

Comment (emackey) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md]: Done.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+clearcoatBlendFactor = clearcoatTexture.r * clearcoatFactor
+clearcoatFresnel = fresnel(0.04, NdotV)
+
+color = mix(vec3(1.0), clearcoat_tint, clearcoatBlendFactor) * (f_emissive + f_diffuse + f_specular) * (1.0 - clearcoatBlendFactor * clearcoatFresnel) +
+        f_clearcoat * clearcoatBlendFactor
+```
+
+## Schema
+
+- [glTF.ADOBE_materials_clearcoat_tint.schema.json](schema/glTF.KHR_materials_clearcoat.schema.json)
```
Comment (lexaknyazev) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:78]: Wrong schema link.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+* Requires the `KHR_materials_clearcoat` extension to also be defined on the material
+
+## Exclusions
+
+* This extension must not be used on a material that also uses `KHR_materials_pbrSpecularGlossiness`.
+* This extension must not be used on a material that also uses `KHR_materials_unlit`.
```
Comment (lexaknyazev) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:19]: Maybe change `*` to `-` to match #2315?

**Review by lexaknyazev — APPROVED** [review]

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+clearcoatBlendFactor = clearcoatTexture.r * clearcoatFactor
+clearcoatFresnel = fresnel(0.04, NdotV)
+
+color = mix(vec3(1.0), clearcoat_tint, clearcoatBlendFactor) * (f_emissive + f_diffuse + f_specular) * (1.0 - clearcoatBlendFactor * clearcoatFresnel) +
+        f_clearcoat * clearcoatBlendFactor
+```
+
+## Schema
+
+- [glTF.ADOBE_materials_clearcoat_tint.schema.json](schema/glTF.ADOBE_materials_clearcoat_tint.schema.json)
```
Comment (lexaknyazev) [inline extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:78]: Should this be aligned with #2313?

Comment (emackey) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:78]: It looks like those changes were only applied to KHR_ extensions, not vendor extensions.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json [hunk 8]
```diff
@@ -0,0 +1,29 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema",
+    "title": "ADOBE_materials_clearcoat_tint glTF extension",
```
[the remark this answers is not in the document]
Comment (emackey) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json:3]: It looks like those changes were only applied to KHR_ extensions, not vendor extensions.

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md (hunk 7 above)
Comment (lexaknyazev) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md:78]: That PR indeed updated only KHR extensions. Most likely one of older KHR files was used as a template for this extension so the naming consistency changes are applicable here as well.

Comment (emackey) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/README.md]: Done

File: extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (emackey) [reply extensions/2.0/Vendor/ADOBE_materials_clearcoat_tint/schema/glTF.ADOBE_materials_clearcoat_tint.schema.json]: Done

**Review by lexaknyazev — APPROVED** [review]

**Pull request merged**